### ISSUE TITLE
Design and print automated turntable vial rack

### ISSUE THREAD

**Issue by sgbaird** [AccelerationConsortium/ac-training-lab#15]
![image](https://github.com/user-attachments/assets/b0b71fb1-446d-4d4f-bbf2-a12761173412)

https://www.tindie.com/products/fluxgarage/turntable-for-stepper-motor-kit/

For automated storage of [Pioreactor vials](https://pioreactor.com/en-ca/collections/accessories-and-parts/products/20ml-glass-vial-closed-cap-and-stir-bar?variant=42836211761208), hopefully some way to make it easier to robotically insert and remove the vials. Maybe some kind of flexible mechanism?

Probably ~8 vials in a single rack. Ideally a Fusion 360 design with parametric modeling.

Something that can be fixtured to the top plate with minimal hardware

Perhaps upload to BambuLab maker world (?)

Aside: vial rack should perhaps have a spot on the side and/or on top for an AprilTag, and numbering of the vial positions along the side (1 through 8).

**Comment by sgbaird** [comment]
@MiceeNS thanks for your initial work on this.

**Comment by sgbaird** [comment]
![image](https://github.com/user-attachments/assets/b07b96a8-e03a-43c1-9777-5bbf0e22948a)

**Comment by MiceeNS** [comment]
Do you have dimensions spec sheet because I cant see find any in the link? Do you already have the automated turntable that is linked above? Are you trying to print only the white storage thing? 

Will update the drawing as fast as possible

**Comment by sgbaird** [comment]
Ah got it, I think the design files might be around somewhere. Yes, I do have a few of these units. One is set up fully.

**Comment by sgbaird** [comment]
I would just be printing whatever attaches to the black top plate.

**Comment by sgbaird** [comment]
@cyrilcaoyang has also been looking into something similar.

**Comment by sgbaird** [comment]
This should have dimensions and design files for the base system: https://www.instructables.com/Automated-Turntable-With-Steppermotor/

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by MiceeNS** [comment]
Thank you, will take a look

**Comment by MiceeNS** [comment]
![Screenshot 2024-10-20 140950](https://github.com/user-attachments/assets/8e7b37ab-c404-48b8-b596-2e2be20fb348)

I think its best to order longer screws for the upper plate and attach the 3d printed vial rack on those screws. Do you want to change something on the vial rack? I suggest printing only one piece of the holder so we are sure fit tight and works well before printing the whole rack.

![Screenshot 2024-10-20 141839](https://github.com/user-attachments/assets/b6fe0d36-2d0d-448a-b4b2-8d75aed5a147)

When I get more time, I will improve some things on the rack

**Comment by sgbaird** [comment]
Overall nice! Thank you. Do you mind starting one of the prints on the Bambu Lab printer?

Everything including and below the yellow plate is something that comes in the kit (except for stepper motor, which we buy separately). The only thing that is 3d printed is the actual rack. Originally, the intention was to have the vial rack attach via the holes on the side.

<img src=https://github.com/user-attachments/assets/537c1893-954a-4eb2-a572-641889b7344e width=250>


(Self-note / aside - file is https://a360.co/3UpCiPU)

**Comment by MiceeNS** [comment]
Hey Sterling,

I will adjust so you can attach it with bolts on those holes. Will also specify which ones. For the Bambu Lab, I have the app but the credentials for log in is asking for email. I cant log in with the credentials you sent me. I made my own account but I am not connected to any device printer. 

Could you please help me out with this.

**Comment by sgbaird** [comment]
That was my bad - just messaged you about the credentials.

**Comment by MiceeNS** [comment]
Thank you, now I am in. I will print it asap. Do you prefer bolts or plastic studs in those holes. Maybe plastic studs would be much rational than bolting the rack?

**Comment by sgbaird** [comment]
> Thank you, now I am in. I will print it asap. Do you prefer bolts or plastic studs in those holes. Maybe plastic studs would be much rational than bolting the rack?

Plastic studs sound fine! Could you include a link to a corresponding mcmaster part, if possible?

**Comment by MiceeNS** [comment]
The plastic studs would be printed with the rack, cylindrical pins that would go in the holes of the turntable. That way you don't need to bolt it, but it can be lifted up.

**Comment by sgbaird** [comment]
Ah, got it! From a 3d printing standpoint, this might change what supports are needed. Considering the small size of the holes, the studs might break off (especially if they're long). Printing in the CF-imbued filament would help a bit with this. I'm not decided one way or the other. Thoughts?

**Comment by MiceeNS** [comment]
Yes absolutely, I was thinking the same It would need to create a lot of supports because those studs are going to lift the rack from the ground. Lets go with bolt and nut. For future we might change and create more generative design.

**Comment by sgbaird** [comment]
Sounds good!

**Comment by MiceeNS** [comment]
Please take a look : https://a360.co/3UpCiPU

Here are the materials you need from McMaster Carr:
https://www.mcmaster.com/91292a113/  x4 
https://www.mcmaster.com/93625A100/ x4 

You can also get any M3 bolt 10mm length and a M3 Nut with compatible thread if you want to go cheaper. I had to make the clearance for the hole close fit since you already have the holes on the Upper plate.

I will send it for print now if you are okey with this.

![Screenshot 2024-10-24 172004](https://github.com/user-attachments/assets/98bb2ab1-b734-4249-94b4-06c5b175e7ed)

**Comment by sgbaird** [comment]
Ok, great! Yes, please do. Just have a look at the print chamber to make sure nothing is in it first

**Comment by sgbaird** [comment]
Oh, actually you might not be able to print today. Looks like someone has an overnight print they're doing.

**Comment by MiceeNS** [comment]
I see printing going at the moment, will take a look later. If not, is it okey for tomorrow?

**Comment by sgbaird** [comment]
👍

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by sgbaird** [comment]
I don't think the print will be removed tonight, so best to wait until tomorrow when it's been taken off. Maybe send the Bambu 3mf file over? (Not sure if there's a way to store the print in a queue)

**Comment by MiceeNS** [comment]
![Screenshot 2024-10-24 173638](https://github.com/user-attachments/assets/456dd835-d81c-4c06-a1e4-52aa672e03fe)

It is higher than the printer can do probably, do you want me to shorten it?

**Comment by sgbaird** [comment]
Yes that sounds good. Maybe adjust for a 20 mL vial?

**Comment by Neil-YL** [comment]
Hi @MiceeNS I was using the 3D printer extensively yesterday. Apologies for any inconvenience.

I opened the 3mf file you sent and the printer generated the supports like this:

<img width="475" alt="image" src="https://github.com/user-attachments/assets/bcdbfa9a-bf23-477e-9bd2-01532e163e23">

Should it be printed this way, or should I adjust the support settings?

**Comment by MiceeNS** [comment]
> Hi @MiceeNS I was using the 3D printer extensively yesterday. Apologies for any inconvenience.
> 
> I opened the 3mf file you sent and the printer generated the supports like this:
> 
> <img alt="image" width="475" src="https://private-user-images.githubusercontent.com/179746567/380196378-bcdbfa9a-bf23-477e-9bd2-01532e163e23.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4NjU1ODUsIm5iZiI6MTcyOTg2NTI4NSwicGF0aCI6Ii8xNzk3NDY1NjcvMzgwMTk2Mzc4LWJjZGJmYTlhLWJmMjMtNDc3ZS05YmQyLTAxNTMyZTE2M2UyMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNVQxNDA4MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYjYxNzZhNzA3YWEwOTM1N2VjNzQxMGI4ODRhN2E2MTE2ZGQyZWYyYWExNmFlMmNiNmVhZGYxZjBhODY5NTNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dgOQllK_ABl4EdIs3mN5qL1pNpx1JHJJNy3Ny1memyw">
> Should it be printed this way, or should I adjust the support settings?

I changed to the normal supports settings, if you think this might cause any issues please adjust. Thank you

**Comment by Neil-YL** [comment]
> > Hi @MiceeNS I was using the 3D printer extensively yesterday. Apologies for any inconvenience.
> > I opened the 3mf file you sent and the printer generated the supports like this:
> > <img alt="image" width="475" src="https://private-user-images.githubusercontent.com/179746567/380196378-bcdbfa9a-bf23-477e-9bd2-01532e163e23.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4NjU1ODUsIm5iZiI6MTcyOTg2NTI4NSwicGF0aCI6Ii8xNzk3NDY1NjcvMzgwMTk2Mzc4LWJjZGJmYTlhLWJmMjMtNDc3ZS05YmQyLTAxNTMyZTE2M2UyMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNVQxNDA4MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yYjYxNzZhNzA3YWEwOTM1N2VjNzQxMGI4ODRhN2E2MTE2ZGQyZWYyYWExNmFlMmNiNmVhZGYxZjBhODY5NTNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dgOQllK_ABl4EdIs3mN5qL1pNpx1JHJJNy3Ny1memyw">
> > Should it be printed this way, or should I adjust the support settings?
> 
> I changed to the normal supports settings, if you think this might cause any issues please adjust. Thank you

I’m not familiar with the purpose of this part, so I’m just double-checking that these supports won’t interfere with its use. 
I’ll proceed with the print as you’ve set it.

**Comment by MiceeNS** [comment]
Are you placing the print in que or you have to wait for the printing to finish?

**Comment by sgbaird** [comment]
> Hi @MiceeNS I was using the 3D printer extensively yesterday. Apologies for any inconvenience.
> 
> I opened the 3mf file you sent and the printer generated the supports like this:
> 
> <img width="475" alt="image" src="https://github.com/user-attachments/assets/bcdbfa9a-bf23-477e-9bd2-01532e163e23">
> 
> Should it be printed this way, or should I adjust the support settings?
> 

Let's hold off on you doing the print. Didn't realize the supports would be quite like this. @Neil-YL can you make sure the print bed is clear of parts at the end of the day? That way @MiceeNS can tune the design and supports as needed and send the print remotely.

Sorry for the back and forth and confusion.

**Comment by Neil-YL** [comment]
> @Neil-YL Are you placing the print in que or you have to wait for the printing to finish?

I have not put it in queue but the printer is available now if you want to access

**Comment by MiceeNS** [comment]
Supports are 30% of the time, the only way I can make this better is changing the actual design of the model. Or make this holders to be printed separate and attached to the column. What do you think @sgbaird ?
![Screenshot 2024-10-25 113319](https://github.com/user-attachments/assets/0751568b-e1b2-4f30-ad3f-02a335457807)

**Comment by sgbaird** [comment]
Ooh, good point about the holders being separate. From a print standpoint, that would make things a lot easier, and a lot better and that it could accommodate a range of vial sizes by inserting different sized holders as needed, and would be relatively straightforward to switch from one vial size to another vial size completely.

If using a mechanism where it clips in to the rack, then as long as there's access from the top for someone's hand to fit through, I think the clip mechanism could be undone.

An alternative (that I don't like as much) is splitting the vial rack in half, printing separately, and attaching the two halves later either with hardware or a "clip together" mechanism.

**Comment by MiceeNS** [comment]
No need for clips (they can be hard to design and easy to break), I would make like a turbine blade root which can be attached to the column. Do you know the diameter and approximate height on 20ml vial? Different vials manufacturer have different diameter on google.
![Screenshot 2024-10-25 115522](https://github.com/user-attachments/assets/8fd5dba4-c258-416a-96c9-b5fd04fd4d50)

I am having a fever at the moment, but will try to make it by tonight and print it.

**Comment by sgbaird** [comment]
Ok, turbine sounds good (makes sense about clips). Take some time to feel better - this one isn't so urgent.

**Comment by MiceeNS** [comment]
Share file link - https://a360.co/3YpRoWC
The triangles are to reduce material and not create supports. The model now is without supports. The holders have same "adapter" so you could just draw another model tomorrow and just attach it. some of the important dimensions you could check them on the drawing. 

Same bolts and nuts like in the previous design.

Let me know what you think and if OK do you want me to print it?
![Screenshot 2024-10-26 164419](https://github.com/user-attachments/assets/fff55f05-08f8-41e3-bd4b-1d60996ed198)
![Screenshot 2024-10-26 165851](https://github.com/user-attachments/assets/8726d9a6-9f51-4657-bf7b-9587f98d939b)
![Screenshot 2024-10-26 174728](https://github.com/user-attachments/assets/59cda412-429a-43cb-92d1-2187b8f79855)

**Comment by sgbaird** [comment]
Way cool! 😎 Yes, please do!

**Comment by MiceeNS** [comment]
![Screenshot 2024-10-26 202914](https://github.com/user-attachments/assets/dac1bf25-6e20-46ab-ba38-2966f96110bd)

![Screenshot 2024-10-26 203216](https://github.com/user-attachments/assets/6a7f8fa5-2d41-409d-8710-280226dc6ab3)


 
There is a problem with the first layer sticking on the plate. I will stop this print, if somebody has a chance to clean the print bed. 

I check on the internet it is some problem with adhesion with the printing bed, it needs to be cleaned with 70% Isopropyl Alcohol. Do you have any suggestions?

**Comment by sgbaird** [comment]
@Neil-YL ?

**Comment by cyrilcaoyang** [comment]
Usually alcohols can help remove the grease possibly from skin contact. If glue was used in the previous print, it is best to wash thoroughly with water and wipe dry with some isopropanol.We recently found in SDL2 that the PEI bed can wear out over extended use and become less sticky to PLA. Hope this is not the case :)YangOn Oct 26, 2024, at 20:36, Bojan Micevski ***@***.***> wrote:﻿
Screenshot.2024-10-26.202914.png (view on web)
Screenshot.2024-10-26.203216.png (view on web)
There is a problem with the first layer sticking on the plate. I will stop this print, if somebody has a chance to clean the print bed.
I check on the internet it is some problem with adhesion with the printing bed, it needs to be cleaned with 70% Isopropyl Alcohol. Do you have any suggestions?

**Comment by Neil-YL** [comment]
> @Neil-YL, have you dealt with this before? (We can address next week)

Based on my experience, I would suggest enabling the Raft layer function when printing parts with very small details on the first layer or when the geometry is not perpendicular to the X/Y axis of the printer.

**Comment by MiceeNS** [comment]
> > @Neil-YL, have you dealt with this before? (We can address next week)
> 
> Based on my experience, I would suggest enabling the Raft layer function when printing parts with very small details on the first layer or when the geometry is not perpendicular to the X/Y axis of the printer.
> 
> <img alt="image" width="410" src="https://private-user-images.githubusercontent.com/179746567/380461269-e6150af3-e345-4609-ba3e-ec463facffbc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5OTIwMDAsIm5iZiI6MTcyOTk5MTcwMCwicGF0aCI6Ii8xNzk3NDY1NjcvMzgwNDYxMjY5LWU2MTUwYWYzLWUzNDUtNDYwOS1iYTNlLWVjNDYzZmFjZmZiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyN1QwMTE1MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ODg0MDE4ZmRhYmEzMzc5NWNlZGY0YmZhZjY2MGE2YmY2ZTk1OGRkY2UyMjIxNTJlNDEzYmNiNzYyNmIzNGE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZrcJ0BIg-hNkHP4VZIsxgOlRFbPRR6td6cjBzv7cw6w">

Thank you for sharing your knowledge! Will do that in the next print.

**Comment by Neil-YL** [comment]
I have cleaned up the print bed

**Comment by MiceeNS** [comment]
> I have cleaned up the print bed 

Thank you, when I get home I will give it another try.

**Comment by MiceeNS** [comment]
I had the same problem today with the print. When you get a chance could you please clean the printing bed @Neil-YL?

Also, is it possible for me to send you the file so you can try to print it? I tried multiple times on different places on the printing bed.  This is only the raft being printed.
![Screenshot 2024-10-28 211342](https://github.com/user-attachments/assets/1161ce10-3112-4aca-b10b-bc98382e7df7)

**Comment by sgbaird** [comment]
@Neil-YL that would be great if you could try to get the file printed. Not a super urgent task - sometime this week or next would be fine.

**Comment by Neil-YL** [comment]
> I had the same problem today with the print. When you get a chance could you please clean the printing bed @Neil-YL?
> 
> Also, is it possible for me to send you the file so you can try to print it? I tried multiple times on different places on the printing bed. This is only the raft being printed. ![Screenshot 2024-10-28 211342](https://private-user-images.githubusercontent.com/76826784/380936578-1161ce10-3112-4aca-b10b-bc98382e7df7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyMDg4NzgsIm5iZiI6MTczMDIwODU3OCwicGF0aCI6Ii83NjgyNjc4NC8zODA5MzY1NzgtMTE2MWNlMTAtMzExMi00YWNhLWIxMGItYmM5ODM4MmU3ZGY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDEzMjkzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzYTJmMGRhMmMyNzEzMTYxOWJlZTdkYmQ1ZTBiNDNhZTE3MGQyYzY4ZTJkMGQxMDI0ZDIzOWQ4MWRhNDNhNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t0iB-5ct_2Gx6rRKuh-elf3kJYKqM6MJIwFMiHSQ8HM)

Of course. Please send the model file to Yanghuang.liu92@Gmail.com

**Comment by sgbaird** [comment]
@MiceeNS you can also attach directly here (not sure if you'd need to zip it based on filetype). If .zip or the file type isn't supported, email is fine too

**Comment by MiceeNS** [comment]
Sent him the file through gmail, luckily I remembered to send it to myself this morning.

**Comment by Neil-YL** [comment]
This is a photo of a failed first layer:
<img width="525" alt="image" src="https://github.com/user-attachments/assets/55206df0-9e6d-4c8e-a020-ec8b7a9cfaac">

The failure mainly located close to the edge of the bed, I am suspecting an uneven temperature distribution of the printing bed. 

After several trails and errors, I seem to have a promising first layer print:

<img width="967" alt="image" src="https://github.com/user-attachments/assets/6e5594f4-8711-49e8-88e3-13227d6c9c91">

What I change to have this print:

1. Initial layer height:  The default setting for Banbu is 50% of the nozzle diameter, which is 0.2 in our case. I changed it to 0.35mm (close to the diameter of the nozzle) to have a better adhesive performance of the raft layer (since we print with raft layer the first layer thickness may effect the quality of the printed parts)
2. Bed temperature: the previous setting is 55 Celsius,  I increase to 65 for the first layer (actually I didn't change it for the whole printing work).
3. I also rearranged the parts to have a more regular placement on the bed and remove the largest part to be print in separate work.

If this print still fails, I would reduce the amount of the parts and place them in the middle of the bed.


Reference:
https://overture3d.com/pages/help-center?hcUrl=%2Fen-US%2Fnot-sticking-to-the-bed--bed-adhesion-issue-765548

**Comment by Neil-YL** [comment]
<img width="680" alt="image" src="https://github.com/user-attachments/assets/01242b90-26a4-437e-afe4-7bb75453ab36">
Could you please help me confirm the number of these two parts, now I have eight for each.

**Comment by MiceeNS** [comment]
8 of each, i dont care about the surface inside the legs , sry I am not responding fast 

Thank you

**Comment by Neil-YL** [comment]
<img width="558" alt="image" src="https://github.com/user-attachments/assets/c0b805c8-2b08-4178-be69-53ad5c9de86d">
<img width="921" alt="image" src="https://github.com/user-attachments/assets/084cb331-c0d1-479a-8ce3-2d91c3e4f62a">
The parts would be ready to use by tomorrow

**Comment by MiceeNS** [comment]
Thank you @Neil-YL , you are the hero!

**Comment by Neil-YL** [comment]
<img width="516" alt="image" src="https://github.com/user-attachments/assets/69ecd75f-5a60-4811-b568-9d5d8c75189e">

**Comment by MiceeNS** [comment]
> <img width="516" alt="image" src="https://github.com/user-attachments/assets/69ecd75f-5a60-4811-b568-9d5d8c75189e">
> 
> 

How does the connections fit? Are they tight? I would not want them the move and also easy to put them in / take them out. 

What are your impressions?

Thank you again for what you are doing.

**Comment by sgbaird** [comment]
@Neil-YL maybe a quick video of insertion and removal?

**Comment by sgbaird** [comment]
The grip on the vial when inserted is pretty good. It could be a bit looser.

@Neil-YL maybe tomorrow you could also upload a video of inserting and removing a vial?

**Comment by MiceeNS** [comment]
> The grip on the vial when inserted is pretty good. It could be a bit looser.
> 
> @Neil-YL maybe tomorrow you could also upload a video of inserting and removing a vial?

we can adjust that no problem, I had no idea of the diameter of the vials and also it depends on the plastic flexibility

**Comment by MiceeNS** [comment]
> The connection is very tight. I have to polish a bit to assemble them in place. It is almost impossible to remove without tools now.
> 
> Here is a short video
> 
>  IMG_0381.MOV

I dont think we want to move them frequently, just in case one breaks to replace it.

**Comment by sgbaird** [comment]
I was impressed it happened to be such a good size! I think somewhat looser on the vial and slightly looser on the inserts and we're golden.

**Comment by Neil-YL** [comment]
Here is a video of inserting and removing a vial

https://github.com/user-attachments/assets/b3a6f911-6654-45e7-ab9e-adbdd0e1c3b6

**Comment by MiceeNS** [comment]
[Holder v2.zip](https://github.com/user-attachments/files/17655051/Holder.v2.zip)
STL file 

I changed the tolerances and the holder grip, it should be better now. I can try printing it if you want when the printing bed is empty. It is only the holder not the whole design

**Comment by sgbaird** [comment]
@Neil-YL could you send the print? @MiceeNS, you're welcome to (seems open now), but there are some other users so occasionally there will be overnight prints.

**Comment by Neil-YL** [comment]
> @Neil-YL could you send the print? @MiceeNS, you're welcome to (seems open now), but there are some other users so occasionally there will be overnight prints, which is when I'd imagine you're usually sending prints.

Sorry I am not in the office this afternoon and it seems the printer is in use now. I can try tomorrow morning

**Comment by Neil-YL** [comment]
<img width="629" alt="image" src="https://github.com/user-attachments/assets/f0134170-ed12-491a-ab0a-d5231066a74a">
Sent to print

**Comment by sgbaird** [comment]
All eight is fine - not a super long print. Maybe if we have future iterations just one so we can check the tolerance and fit first.

**Comment by Neil-YL** [comment]
The new design is too loose to hold the vial. I should have printed once for trial first.

Here are the videos:

https://github.com/user-attachments/assets/8046cd1d-7cdd-49bc-8a61-28652c5d09f7

https://github.com/user-attachments/assets/0547bf45-066f-4cee-b522-dea6a9455c0b

**Comment by sgbaird** [comment]
Thanks @Neil-YL!

@MiceeNS, I'm thinking of two options:

1. We add a little platform for the vial to sit on (I had been considering this, and am thinking now maybe it would be worthwhile)
2. We switch to a single-layer design that is basically just 8 holes, maybe with a slight backdrop to each one

**Comment by MiceeNS** [comment]
[Holder v3.zip](https://github.com/user-attachments/files/17788239/Holder.v3.zip)
@Neil-YL I am printing another holder, could you please test it out when you get there. Also, would you be kind to give me the dimensions of the vial like diameter and height 
![image](https://github.com/user-attachments/assets/2773ed12-5254-4a42-8963-d196a4a3ab4e)

Even though we might change the complete design I do not want to leave this unfinished.

**Comment by Neil-YL** [comment]
> [Holder v3.zip](https://github.com/user-attachments/files/17788239/Holder.v3.zip) @Neil-YL I am printing another holder, could you please test it out when you get there. Also, would you be kind to give me the dimensions of the vial like diameter and height !

The diameter of the vial is 27.5mm and height(with cap) is 66mm

The new holder is way too large compared with previous two, here is a video testing the vial holding with three designs:

https://github.com/user-attachments/assets/ba91dfb1-b1ae-4b86-b6b1-e656912f0187

**Comment by MiceeNS** [comment]
> > [Holder v3.zip](https://github.com/user-attachments/files/17788239/Holder.v3.zip) @Neil-YL I am printing another holder, could you please test it out when you get there. Also, would you be kind to give me the dimensions of the vial like diameter and height !
> 
> The diameter of the vial is 27.5mm and height(with cap) is 66mm
> 
> The new holder is way too large compared with previous two, here is a video testing the vial holding with three designs:
> 
>  IMG_0694.mov

Thank you

**Comment by sgbaird** [comment]
Maybe worth printing a series of sizes?

**Comment by Neil-YL** [comment]
> Maybe worth printing a series of sizes?

Should I test the third one (the most recently printed) with a different size vial? I’m not very familiar with this project, make sure I’m not missing any context

**Comment by MiceeNS** [comment]
> > Maybe worth printing a series of sizes?
> 
> 
> 
> Should I test the third one (the most recently printed) with a different size vial? I’m not very familiar with this project, make sure I’m not missing any context

If you have time it's not gonna hurt to try it but initially i was designing it for the same vial.I had no idea on the dimensions and was just guessing. Next I will print 3-4 designs hopefully it will work.

What you can try is the last design tolerances on the rack. See if it fits nice.

Thank you for all your help, I don't know how to make it up for everything you do!

**Comment by Neil-YL** [comment]
I think the fit on the rack is excellent. Stable and not difficult to remove. 

Here is the video:

https://github.com/user-attachments/assets/ebb3814d-c569-4772-b794-b093c784f598

I’m not sure if we have other sizes of vials in the office. Maybe Sterling could let me know if you’d like to try designing a series with different sizes.

**Comment by MiceeNS** [comment]
From the video Design 3 is the best in my opinion. @sgbaird let me know if you want me to print 8 of them for the rack. Or maybe there is still room for improvement on the holder

**Comment by sgbaird** [comment]
Thanks @Neil-YL! The video was really informative. Thanks for the designs @MiceeNS! I agree, let's go with design 3. We can wait on printing more until we do some testing.

@MiceeNS could you tell us the dimensions of the square plate meant for the AprilTag?

@SissiFeng, could you print an AprilTag of the dimension @MiceeNS provides and put it in the training lab office? I'll leave it up to you which family and tag you want to use.

@Neil-YL could you fixture the AprilTag to the vial rack?

@gursi26 can you try out a vial insertion and removal workflow with a MyCobot 280 Pi? (fyi to Bojan: it's a low-cost robot arm, but also lower accuracy/precision and less force).

A picture of the vial rack with the fixtured AprilTag/holder and videos of the insertion/removal attempts (even and especially failed attempts) would be great to have.

Once we have a setup working decently, then I suggest we print 8 holders of the final design (might be the same version as above), and I'll ask someone to handle the stepper motor of the turntable.

**Comment by MiceeNS** [comment]
@SissiFeng the square dimensions are **25mm x 25mm**. Picture of the square plate


![Capture](https://github.com/user-attachments/assets/fd8dc595-8fd0-42e0-a879-ba540bec23e2)

**Comment by Neil-YL** [comment]
@sgbaird Sure. I will get the vial rack ready for MyCobot test.

@MiceeNS Do you need me to print the new vial holder? Someone else is using the printer (until 14:00) I can set up the print after that one finish if you can send me the model file.

**Comment by MiceeNS** [comment]
> @sgbaird Sure. I will get the vial rack ready for MyCobot test.
> 
> 
> 
> @MiceeNS Do you need me to print the new vial holder? Someone else is using the printer (until 14:00) I can set up the print after that one finish if you can send me the model file. 

I do not have Design 3 with the root that can be mounted on the rack. I will do that tonight and send one to print.

**Comment by sgbaird** [comment]
Root, as in a little shelf at the bottom?

**Comment by MiceeNS** [comment]
> Root, as in a little shelf at the bottom?

That is the word that I came up with :) The last designs did not had the feature to be attached on the rack. I did not want to waste material if the designs did not work.
The green marked on the picture: 
![Capture](https://github.com/user-attachments/assets/d19bea00-3f13-4cd0-9ad3-813e106d92c9)

**Comment by sgbaird** [comment]
Oh, got it! The "puzzle piece" portion at the base of the clip. I was thinking about a small shelf for the vial to sit on to keep a more consistent XYZ position for the vial. Thanks for clarifying!

**Comment by MiceeNS** [comment]
Oh, you mentioned that before. I could add like a bottom support for the vial to sit on. Whoever the holder grasp the vial and it will just levitate from the bottom support. Unless it should be like a slot that guides the vial to the holder position.

**Comment by sgbaird** [comment]
> I could add like a bottom support for the vial to sit on

That would be great. Not required for the tests I described above, but maybe nice to have.

**Comment by Neil-YL** [comment]
Hi @MiceeNS 
Someone plans to print a large model over the weekend (but it might not start before your tonight). Since no one can collect our part during the weekend to clean up the plate for them, I think you can post the model here after you finish the designing, and I’ll print it as soon as I get to the office next Monday. 

I will print one first for test and if everything works well I will do eight then.

**Comment by MiceeNS** [comment]
Here are two designs. There is nothing on bambu studio currently on the printing bed so I could print them. The design with the bottom support requires bunch of supports i can adjust it in future for less supports. Let me know if you want me to print both design

Attaching zip file with both STL files and the bambu lab project.
[Holder Designs.zip](https://github.com/user-attachments/files/17883020/Holder.Designs.zip)

Shared file link: https://a360.co/3V1MVc0

Pictures:
![Screenshot 2024-11-23 175758](https://github.com/user-attachments/assets/6e55d360-c471-43ce-9727-fba78bd91a17)
![Screenshot 2024-11-23 175216](https://github.com/user-attachments/assets/a18d9943-05b1-4091-b2d8-5055ae5cccfe)
![Screenshot 2024-11-23 174742](https://github.com/user-attachments/assets/04beeb80-9656-4bb2-87c8-e33d6e89bade)
![Screenshot 2024-11-23 174705](https://github.com/user-attachments/assets/9109b945-d052-46ca-9cef-299ec1646d1a)
![Screenshot 2024-11-23 174613](https://github.com/user-attachments/assets/ee612ee1-7899-4243-9a93-47f4024d7702)

**Comment by sgbaird** [comment]
I think the idea is that someone is planning to start a print tomorrow that may take up most of Saturday and Sunday, so if you haven't submitted the job already, maybe best to wait for @Neil-YL to run it on Monday. If you have, no worries ☺️ training lab projects like these take priority since it's owned by the training lab.

**Comment by sgbaird** [comment]
Cool designs!

**Comment by MiceeNS** [comment]
I will try to redesign this, now having a full picture of the automation and the purpose of rack I think there are simpler and better ways of doing it.

**Comment by MiceeNS** [comment]
@sgbaird  I created something else and tried to do quick animation. Fusion 360 is not the best for animations but still for representation

Shared Fusion file: https://a360.co/40iKq6L

https://github.com/user-attachments/assets/0fb02dde-2a71-425d-b0f9-aca263fad943

![Screenshot 2025-01-16 003246](https://github.com/user-attachments/assets/6cb2ddad-0b30-4881-a4dd-72694a127b2b)

The labeled green things are for guiding the vial to position and they do not create any force on the vial while in place
![Screenshot 2025-01-16 003433](https://github.com/user-attachments/assets/ccdb6b10-b31e-44a5-8c80-2c950980be94)

**Comment by sgbaird** [comment]
Nice! Let's try this out (just seeing now). Could you send the print to the X1E at some point? I think we switched to a 0.6 mm nozzle.

**Comment by MiceeNS** [comment]
> Nice! Let's try this out (just seeing now). Could you send the print to the X1E at some point? I think we switched to a 0.6 mm nozzle.

Sure, will take a look in the next days and print it out. Will notify you

**Comment by MiceeNS** [comment]
It happened again with the print bed. @Neil-YL  can you print it out tomorrow? I am attaching the  STL file in ZIP file

[Vial Rack.zip](https://github.com/user-attachments/files/19219914/Vial.Rack.zip)

Thanks

**Comment by Neil-YL** [comment]
> It happened again with the print bed. [@Neil-YL](https://github.com/Neil-YL) can you print it out tomorrow? I am attaching the STL file in ZIP file
> 
> [Vial Rack.zip](https://github.com/user-attachments/files/19219914/Vial.Rack.zip)
> 
> Thanks

I restart the same print job saved on the machine. I will see if anything I need to adjust to improve the quality

**Comment by MiceeNS** [comment]
> > It happened again with the print bed. [@Neil-YL](https://github.com/Neil-YL) can you print it out tomorrow? I am attaching the STL file in ZIP file
> > 
> > [Vial Rack.zip](https://github.com/user-attachments/files/19219914/Vial.Rack.zip)
> > 
> > Thanks
> 
> I restart the same print job saved on the machine. I will see if anything I need to adjust to improve the quality

Thank you, confirm if it is printed OK

**Comment by MiceeNS** [comment]
Last prints I would increase the printing bed temp to 65 but this time did not worked

**Comment by sgbaird** [comment]
@Neil-YL showed me a printed part yesterday (not sure if this is the same one you're referring to). Looked fine to me!

**Comment by sgbaird** [comment]
I think this works pretty well! Positioning seems repeatable (don't worry so much about my "stress test" placements)

https://github.com/user-attachments/assets/97243248-b328-4729-865f-fae4f147504b

Frankly I'm not sure where our turntable is, so I can't verify the hole location directly, but the holes match up with the white one, so should be fine!

**Comment by sgbaird** [comment]
Cc @kelvinchow23, any thoughts here? (See video)

**Comment by kelvinchow23** [comment]
nice looks good! i took a look at it in person and it looks good too. if you find the positioning twigs breaking often, I would look into making that part separate as an insert to the base plate to make it easier to replace. is the plan for the robot arm to drop it at a height similar to that video (a couple of cm)?

**Comment by sgbaird** [comment]
> if you find the positioning twigs breaking often, I would look into making that part separate as an insert to the base plate to make it easier to replace

Let's stress-test the current design a bit (i.e., put it into use for a few weeks / months) and see if they're breaking regularly.

> is the plan for the robot arm to drop it at a height similar to that video (a couple of cm)?

Just for stress testing. A more realistic one might have been for me to push it down from a slightly off-center position and then release.

**Comment by MiceeNS** [comment]
We can also add 4 twigs if the vial falls through them just as you shown on the video of the "stress test"

**Comment by MiceeNS** [comment]
I will adjust that right now, meanwhile I see you are using the AprilTag on the vial caps. You want to create some other "post" for that?

**Comment by sgbaird** [comment]
Thanks! You can ignore the AprilTags on the vials. The idea is more so that this would always be attached to the cap (i.e., sample). This would be separate from the vial holder position.

**Comment by MiceeNS** [comment]
I am testing some joints so we can print them separate, that way instead of printing bottom to top we can print it sideways which will add bit more strength. I will update you...

**Comment by sgbaird** [comment]
Vial rack got toppled over accidentally in https://github.com/AccelerationConsortium/ac-training-lab/issues/270#issuecomment-2965986355, and a few more broke:

![Image](https://github.com/user-attachments/assets/6e411f95-5a62-4cff-a95d-3c1013abede6)

One of the nice things that I noticed is that there is still an inset for the vial to sit in (i.e., like a "typical" vial rack)
